shesha-core/src/Shesha.Application/NotificationMessages/NotificationMessageAppService.cs

```diff
@@ -33,4 +34,18 @@ public async Task<bool> Resend(Guid id)
 
         return true;
     }
+
+    public async Task<DynamicDto<NotificationMessage, Guid>> MarkAsReadAsync(Guid id)
+    {
+        if (id == Guid.Empty)
+            throw new ArgumentNullException(nameof(id));
+
+        var entity = await SaveOrUpdateEntityAsync<NotificationMessage>(id, item =>
```

Comment:
What should happen if the message was already marked as opened previously? Should it overwrite the timestamp? Probably not I think.